cryptarchia/messages.abnf

```diff
@@ -0,0 +1,21 @@
+; VERSION 0.1
+; ------------ BLOCK ----------------------
+BLOCK            = HEADER CONTENT
+; ------------ HEADER ---------------------
+VERSION          = %x01
+HEADER           = VERSION HEADER-UNSIGNED
+HEADER-UNSIGNED  = %x00 HEADER-COMMON
+HEADER-COMMON    = CONTENT-SIZE CONTENT-ID BLOCK-DATE PARENT-ID
+CONTENT-SIZE     = U32
+BLOCK-DATE       = BLOCK-SLOT
```

Comment:
should we use epoch + slot or just absolute slot? any consideration?

Reply:
epoch seems useful to know.

One potential use: say you receive a block, you want to know if it's relevant to you or not right? just knowing the slot and the parent wouldn't be enough to know if your just behind? or if this block is actually from a completely different epoch that we've forgotten about.

Reply:
actually, I take that back I'm looking at the ouroborous byron specification and I don't see an epoch listed. Probably wise to keep things as minimal as we can.
<img width="687" alt="Screenshot 2024-01-31 at 2 27 25 PM" src="https://github.com/logos-co/nomos-specs/assets/1832378/0ab46450-b259-43c6-b2f6-9b0920b9fa24">

Reply:
> just knowing the slot and the parent wouldn't be enough to know if your just behind? or if this block is actually from a completely different epoch that we've forgotten about.

epoch + slot inside the epoch or absolute slot are equivalent in term of information, I feel the latter is a easier to work with for time calculations but a bit more inconvenient for snapshots and such.
Anyway, this is just one format, what is used in the code for other operations is entirely decoupled.